packages/api/src/lib/granule-delete.ts

```diff
@@ -45,13 +52,17 @@ const deleteGranuleAndFiles = async ({
   filePgModel = new FilePgModel(),
   granulePgModel = new GranulePgModel(),
   granuleModelClient = new Granule(),
+}: {
+  knex: Knex,
+  dynamoGranule: ApiGranule,
+  pgGranule: PostgresGranuleRecord,
+  filePgModel: FilePgModel,
+  granulePgModel: GranulePgModel,
+  granuleModelClient: typeof Granule
 }) => {
   if (pgGranule === undefined) {
-    if (dynamoGranule.files && dynamoGranule.files.length > 1) {
-      // Delete only the Dynamo Granule and S3 Files
-      await _deleteS3Files(dynamoGranule.files);
-    }
-
+    // Delete only the Dynamo Granule and S3 Files
+    await _deleteS3Files(dynamoGranule.files);
```

Comment:
Typescript won't complain if you were to pass a value for `files` that is `undefined`, since it will default it to an empty array. But if you were to pass in an array of values that didn't conform to the expected type (`ApiFile | PostgresFileRecord` for each array item), it would complain.